src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -1122,11 +1101,18 @@ internal void FlushContentToDisk()
                     {
                         try
                         {
+                            Encoding currentEncoding = null;
+
+                            using (StreamReader reader = new StreamReader(this.Path, Utils.utf8NoBom, true))
+                            {
+                                currentEncoding = reader.CurrentEncoding;
```

Comment:
You lost a formal read operation to force encoding detection.

Reply:
done